source/MaterialXView/OpenGL/GLUtilityContext.cpp

```diff
@@ -324,37 +308,23 @@ int GLUtilityContext::makeCurrent()
 #elif defined(OSLinux_)
     makeCurrentOk = glXMakeCurrent(_display, _dummyWindow, _contextHandle);
 #elif defined(OSMac_)
-    aglToNSOpenGLMakeCurrent(_contextHandle);
-    if (aglToNSOpenGLGetCurrentContextWrapper() == _contextHandle)
+    NSOpenGLMakeCurrent(_contextHandle);
+    if (NSOpenGLGetCurrentContextWrapper() == _contextHandle)
     {
         makeCurrentOk = 1;
     }
-    aglCheckError();
 #else
     ;
 #endif
     return makeCurrentOk;
 }
 
 //
-// Singleton create/destory methods
+// Creator
 //
-GLUtilityContext* GLUtilityContext::create(const WindowWrapper& windowWrapper, HardwareContextHandle context)
-{
-    if (!_globalGLUtilityContext)
-    {
-        _globalGLUtilityContext = new GLUtilityContext(windowWrapper, context);
-    }
-    return _globalGLUtilityContext;
-}
-
-void GLUtilityContext::destroy()
+GLUtilityContextPtr GLUtilityContext::creator(const WindowWrapper& windowWrapper, HardwareContextHandle context)
 {
-    if (_globalGLUtilityContext)
-    {
-        delete _globalGLUtilityContext;
-        _globalGLUtilityContext = nullptr;
-    }
+    return std::shared_ptr<GLUtilityContext>(new GLUtilityContext(windowWrapper, context));
```

Comment:
Nitpick: could use make_shared instead of new:

 ```return std::make_shared<GLUtilityContext>(windowWrapper, context); ```

Reply:
The constructor is private so it can't be accessed on purpose so I need to do it this way.